array.go

```diff
@@ -22,7 +23,11 @@ type Array struct {
 func NewArray(reporter Reporter, value []interface{}) *Array {
 	chain := makeChain(reporter)
 	if value == nil {
-		chain.fail("expected non-nil array value")
+		failure := Failure{
+			err:        fmt.Errorf("expected non-nil array value"),
+			assertType: failureAssertNotNil,
+		}
+		chain.fail(failure)
```

Comment:
Nitpick: is there a point of writing this instead of

```
chain.fail(Failure{
    err:        fmt.Errorf("expected non-nil array value"),
    assertType: failureAssertNotNil,
})
```

?

Reply:
mh probably because it was a bunch of copypastes and didn't notice :) doing all those changes on all functions kinda turned my brain off :sweat_smile: